### PR TITLE
Initial support and wiring for developer settings

### PR DESCRIPTION
**Pull request by jarolrod**
This introduces initial support and wiring for the currently implemented settings that are part of the `DeveloperSettings.qml`.

[Windows](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/224)
[Intel macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/224)
[Apple Silicon macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/224)
[ARM64 Android](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/224)

### CODE REVIEW

**Comment by hebasto** [comment]
> based on #222

It has been merged. Mind rebasing?

**Comment by jarolrod** [comment]
Updated from a17ab60 to 43ef370

changes: rebased over main

File: src/qml/options_model.h [hunk 1]
```diff
@@ -35,27 +39,33 @@ class OptionsQmlModel : public QObject
...
     void pruneSizeGBChanged(int new_prune_size_gb);
+    void scriptThreadsChanged(int new_script_threads);
     void serverChanged(bool new_server);
     void upnpChanged(bool new_upnp);
 
 private:
     interfaces::Node& m_node;
 
     // Properties that are exposed to QML.
+    bool m_dbcache_size_mib;
```
Comment (hebasto) [inline src/qml/options_model.h:63]:
```suggestion
    int m_dbcache_size_mib;
```

**Review by hebasto — COMMENTED** [review]
> This does not set defaults for these in regards to the UI, a follow-up will handle already defined defaults.

Why postpone it?

```diff
diff --git a/src/qml/components/DeveloperOptions.qml b/src/qml/components/DeveloperOptions.qml
index 4ad8741fb..fe7b5cece 100644
--- a/src/qml/components/DeveloperOptions.qml
+++ b/src/qml/components/DeveloperOptions.qml
@@ -24,7 +24,7 @@ ColumnLayout {
         header: qsTr("Database cache size")
         actionItem: RowLayout {
             ValueInput {
-                description: "450"
+                description: optionsModel.dbcacheSizeMiB
                 onEditingFinished: optionsModel.dbcacheSizeMiB = parseInt(text)
             }
             Header {
@@ -37,7 +37,7 @@ ColumnLayout {
         Layout.fillWidth: true
         header: qsTr("Script verification threads")
         actionItem: ValueInput {
-            description: "0"
+            description: optionsModel.scriptThreads
             onEditingFinished: optionsModel.scriptThreads = parseInt(text)
         }
     }
diff --git a/src/qml/options_model.cpp b/src/qml/options_model.cpp
index 6f536bb29..268682013 100644
--- a/src/qml/options_model.cpp
+++ b/src/qml/options_model.cpp
@@ -7,18 +7,24 @@
 #include <interfaces/node.h>
 #include <qt/guiconstants.h>
 #include <qt/optionsmodel.h>
+#include <txdb.h>
 #include <univalue.h>
 #include <util/settings.h>
 #include <util/system.h>
+#include <validation.h>
 
 #include <cassert>
 
 OptionsQmlModel::OptionsQmlModel(interfaces::Node& node)
     : m_node{node}
 {
+    m_dbcache_size_mib = SettingToInt(m_node.getPersistentSetting("dbcache"), nDefaultDbCache);
+
     int64_t prune_value{SettingToInt(m_node.getPersistentSetting("prune"), 0)};
     m_prune = (prune_value > 1);
     m_prune_size_gb = m_prune ? PruneMiBtoGB(prune_value) : DEFAULT_PRUNE_TARGET_GB;
+
+    m_script_threads = SettingToInt(m_node.getPersistentSetting("par"), DEFAULT_SCRIPTCHECK_THREADS);
 }
 
 void OptionsQmlModel::setDbcacheSizeMiB(int new_dbcache_size_mib)
```

File: src/qml/options_model.h (hunk 1 above)
Comment (johnny9) [reply src/qml/options_model.h:63]: Apparently clang-tidy can catch these issues. I was surprised gcc doesn't warn about this.

Comment (jarolrod) [reply src/qml/options_model.h]: hehe

**Comment by jarolrod** [comment]
Updated from[ 43ef370 ](https://github.com/bitcoin-core/gui-qml/commit/43ef37004c1412ceccc53219945726a88e675c8c)to[ ce05986](https://github.com/bitcoin-core/gui-qml/commit/ce059860d4b3395b5ad892162dabac8a42b2577d), [compare](https://github.com/bitcoin-core/gui-qml/compare/43ef37004c1412ceccc53219945726a88e675c8c..ce059860d4b3395b5ad892162dabac8a42b2577d)

Changes: addressed review feedback 

@hebasto I hadn't setup defaults because i've been reworking the settings control, but i'll just build upon this.

**Comment by hebasto** [comment]
Rebase?

**Comment by jarolrod** [comment]
Updated from [ce05986 ](https://github.com/bitcoin-core/gui-qml/commit/ce059860d4b3395b5ad892162dabac8a42b2577d)to [3bc5c97](https://github.com/bitcoin-core/gui-qml/commit/3bc5c975ef6c85d72164d24f45f27ab1eab13616), [Compare](https://github.com/bitcoin-core/gui-qml/compare/ce059860d4b3395b5ad892162dabac8a42b2577d..3bc5c975ef6c85d72164d24f45f27ab1eab13616)

changes:
- rebased over main
- Moved the declaration of "MiB" into the setting description for simplicity of state behavior and focus control

**Review by johnny9 — APPROVED** [review]
ACK 3bc5c975ef6c85d72164d24f45f27ab1eab13616

Tested by checking the text fields in developer options and see the appropriate values in "par" and "dbcache" fields in settings.json.

**Pull request merged**